ui/app/clinical/consultation/controllers/addTreatmentController.js

```diff
@@ -812,6 +812,34 @@ angular.module('bahmni.clinical')
                 }
             };
 
+            $scope.$watch('treatment.route', function (newValue, oldValue) {
+                if (newValue !== oldValue) {
+                    $scope.checkForContinuousMedication(newValue);
+                }
+            });
+
+            $scope.$watch('treatment.quantity', function(newValue) {
+                if (newValue === 0 && $scope.isContinuousMedication && !$scope.treatment.uniformDosingType.frequency) {
+                    $scope.treatment.quantity = null;
+                }
+            });
+
+            $scope.checkForContinuousMedication = function (route) {
+                if ($scope.continuousMedicationRoutes.includes(route)) {
+                    $scope.isContinuousMedication = true;
```

Comment:
we can assign $scope.continuousMedicationRoutes.includes(route) to $scope.isContinuousMedication, so that we can avoid unnecessary if else block

Reply:
fixed it